packages/assets-controllers/src/AssetsContractController.ts

```diff
@@ -62,137 +67,225 @@ export const SINGLE_CALL_BALANCES_ADDRESS_BY_CHAINID: Record<Hex, string> = {
     '0x6aa75276052d96696134252587894ef5ffa520af',
   [SupportedTokenDetectionNetworks.moonriver]:
     '0x6aa75276052d96696134252587894ef5ffa520af',
-};
+} as const satisfies Record<Hex, string>;
 
 export const MISSING_PROVIDER_ERROR =
   'AssetsContractController failed to set the provider correctly. A provider must be set for this method to be available';
 
-/**
- * @type AssetsContractConfig
- *
- * Assets Contract controller configuration
- * @property provider - Provider used to create a new web3 instance
- */
-// This interface was created before this ESLint rule was added.
-// Convert to a `type` in a future major version.
-// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
-export interface AssetsContractConfig extends BaseConfig {
-  provider: Provider | undefined;
-  ipfsGateway: string;
-  chainId: Hex;
-}
-
 /**
  * @type BalanceMap
  *
  * Key value object containing the balance for each tokenAddress
  * @property [tokenAddress] - Address of the token
  */
-// This interface was created before this ESLint rule was added.
-// Convert to a `type` in a future major version.
-// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
-export interface BalanceMap {
+export type BalanceMap = {
   [tokenAddress: string]: BN;
-}
+};
+
+const name = 'AssetsContractController';
+
+// TODO: Convert into generic type that takes controller type as input and move to base-controller
+type AssetsContractControllerActionsMap = {
+  [ClassMethod in keyof AssetsContractController as AssetsContractController[ClassMethod] extends ActionConstraint['handler']
+    ? ClassMethod
+    : never]: {
+    type: `${typeof name}:${ClassMethod}`;
+    handler: AssetsContractController[ClassMethod];
+  };
+};
+
+type AssetsContractControllerMethodName =
+  keyof AssetsContractControllerActionsMap;
+
+export type AssetsContractControllerActions =
+  AssetsContractControllerActionsMap[AssetsContractControllerMethodName];
```

Comment:
Proof of concept for new base-controller utility type that would simplify the workflow for creating a new messenger-consumer class, or batch adding a large number of methods/actions.